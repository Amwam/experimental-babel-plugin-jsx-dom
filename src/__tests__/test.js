const babel = require('@babel/core');
const plugin = require('../');

function check(codeExample) {
  const { code } = babel.transform(codeExample, {
    plugins: ['@babel/plugin-syntax-jsx', plugin],
  });

  expect(code).toMatchSnapshot();
}

test('Empty element', () => {
  check('<div />');
});

test('No children', () => {
  check('<div></div>');
});

test('Attributes', () => {
  check('<div attribute="test" attribute2="test2" />');
});

test.skip('Text child', () => {
  check('<div>Hello, world!</div>');
});

test.skip('Text child, with attribute', () => {
  check('<div attribute="testing">Hello, world!</div>');
});

test.skip('JSX children', () => {
  check('<div><span>Hello, world!</span></div>');
});
