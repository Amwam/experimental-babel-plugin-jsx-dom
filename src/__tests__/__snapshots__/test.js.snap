// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attributes 1`] = `"document.createElement(\\"div\\").setAttribute(\\"attribute\\", \\"test\\").setAttribute(\\"attribute2\\", \\"test2\\");"`;

exports[`Empty element 1`] = `"document.createElement(\\"div\\");"`;

exports[`No children 1`] = `"document.createElement(\\"div\\");"`;
